src/lia/util/net/common/NetloggerRecord.java

```diff
@@ -0,0 +1,143 @@
+package lia.util.net.common;
+
+import java.net.InetAddress;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+/**
+ * @author nlmills@g.clemson.edu
+ */
+public class NetloggerRecord {
+    /** the size of the data block read from the disk and posted to the network */
+    private int block;
+    /** tcp buffer size (if 0 system defaults were used) */
+    private int buffer;
+    /** the FTP rfc959 completion code of the transfer */
+    private String code = "429";  // = "Connection closed; transfer aborted."
+    /** time the transfer completed */
+    private Date completed = new Date();
+    /** the destination host */
+    private InetAddress destination = InetAddress.getLoopbackAddress();
```

Comment:
You want to have here null, because if client is transferring non-existing file, this will fail:
```
java.lang.NoSuchMethodError: java.net.InetAddress.getLoopbackAddress()Ljava/net/InetAddress;
	at lia.util.net.common.NetloggerRecord.<init>(NetloggerRecord.java:20)
	at lia.util.net.copy.FDTWriterSession.finalCleanup(FDTWriterSession.java:144)
	at lia.util.net.copy.FDTWriterSession.internalClose(FDTWriterSession.java:329)
	at lia.util.net.common.AbstractFDTCloseable$AsynchronousCloseThread.run(AbstractFDTCloseable.java:85)
```